packages/components/src/generic_modal/generic_modal.tsx

```diff
@@ -143,6 +144,7 @@ export class GenericModal extends React.PureComponent<Props, State> {
                 <h1 id='genericModalLabel'>
                     {this.props.modalHeaderText}
                 </h1>
+                {this.props.headerButton}
```

Comment:
you could add layout related class based on headerButton(context my comment on changing existing css class)

Reply:
Yeah, that makes sense for CSS. Instead, here I am passing a button to display. It is an optional prop, so I dont think it would create any issue. Please let me know, if I am missing something here.